src/pybind/mgr/dashboard/controllers/rgw.py

```diff
@@ -111,6 +111,27 @@ def migrate(self, daemon_name=None, realm_name=None, zonegroup_name=None, zone_n
                                                          secret_key)
         return result
 
+    @RESTController.Collection(method='PUT', path='/setup_multisite_replication')
+    @allow_empty_body
+    # pylint: disable=W0102,W0613
+    def setup_multisite_replication(self, daemon_name=None, realm_name=None, zonegroup_name=None,
+                                    zonegroup_endpoints=None, zone_name=None, zone_endpoints=None,
+                                    username=None, cluster=None):
+        multisite_instance = RgwMultisite()
+        result = multisite_instance.setup_multisite_replication(realm_name, zonegroup_name,
+                                                                zonegroup_endpoints, zone_name,
+                                                                zone_endpoints, username,
+                                                                cluster)
+        return result
+
+    @RESTController.Collection(method='GET', path='/restart_rgw_daemons_and_set_credentials')
```

Comment:
same here. its not really good to call an endpoint `restart_rgw_daemons_and_set_credentials`. We should name it short and clear. and why is this a `GET` endpoint. It is doing some action so it should be different, maybe PUT?